StateMachine.cpp

```diff
@@ -12,7 +12,7 @@ StateMachine::StateMachine(BYTE maxStates, BYTE initialState) :
 	m_pEventData(NULL)
 {
 	ASSERT_TRUE(MAX_STATES < EVENT_IGNORED);
```

Comment:
I ran into a "redefined ASSERT_TRUE" issue when I try to build unit-tests for my state machine. Can we also rename this to avoid that conflict? I change it to something like "SM_ASSERT_TRUE" in my local branch.